pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -43,11 +45,21 @@ const (
 
 var runInTestMode bool
 var runInDemoMode bool
+var maxConcurrentReconciles int
 
 func init() {
 	flag.BoolVar(&runInTestMode, "test-mode", false, "disable ironic communication")
 	flag.BoolVar(&runInDemoMode, "demo-mode", false,
 		"use the demo provisioner to set host states")
+
+	var err error
+	maxConcurrentReconciles, err = strconv.Atoi(os.Getenv("OPERATOR_CONCURRENCY"))
+	if err != nil || maxConcurrentReconciles <= 0 {
+		log.Info("OPERATOR_CONCURRENCY is either unset or invalid. Using default: 3")
```

Comment:
If it contains an invalid value, we should complain loudly about it to ensure someone notices. Someone tried to set a value and we are now ignoring it. I suggest at least logging it as a warning or error, but exiting non-zero would be completely reasonable for an invalid config.

Reply:
I think the new commit is a little better about this. Please take a look.